### PR TITLE
Add enableScenes API in Rainmaker

### PR DESCRIPTION
**Pull request by sanketwadekar**
1. This PR adds a new API to use Rainmaker Scenes in Arduino.
2. I have updated the documentation for the API.
3. I have updated the examples and tested them on esp32.

### CODE REVIEW

**Comment by VojtechBartoska** [comment]
we have some issue with Documentation building CI

File: docs/source/api/rainmaker.rst [hunk 1]
```diff
@@ -127,6 +123,21 @@ This function will return
 1. `ESP_OK` : On success
 2. Error in case of failure
 
+RMaker.enableScenes
+*****************
```
Comment (P-R-O-C-H-Y) [inline docs/source/api/rainmaker.rst:127]: Missing **

File: docs/source/api/rainmaker.rst [hunk 2]
```diff
@@ -13,7 +13,6 @@ ESP RainMaker is an end-to-end solution offered by Espressif to enable remote co
 - RainMaker library (i.e. this library, to develop the firmware)
 - RainMaker Cloud (backend, offering remote connectivity)
 - RainMaker Phone App/CLI (Client utilities for remote access)
-
```
Comment (P-R-O-C-H-Y) [inline docs/source/api/rainmaker.rst:16]: Missing empty line is causing CI docs build to fail @VojtechBartoska. Please fix it @sanketwadekar

File: docs/source/api/rainmaker.rst (hunk 1 above)
Comment (sanketwadekar) [reply docs/source/api/rainmaker.rst]: Done

File: docs/source/api/rainmaker.rst (hunk 2 above)
Comment (sanketwadekar) [reply docs/source/api/rainmaker.rst]: Done

**Comment by tally0highwayman** [comment]
My apologies in advance if I am not following proper GitHub protocol - Is this to resolve this error :
`'class RMakerClass' has no member named 'enableScenes'; did you mean 'enableSchedule' ?`
I copied the latest rainmaker.rst to my library, but still this compile error.
`docs/source/api/rainmaker.rst`

**Comment by P-R-O-C-H-Y** [comment]
@pedrominatel can you take a look at that please?

**Comment by pedrominatel** [comment]
Are you using the latest master, @tally0highwayman?

Make sure you have your Arduino-ESP32 core updated with the master, otherwise you need to wait until the next release to use those new functions.

**Comment by tally0highwayman** [comment]
I do not understand the expression “Arduino-ESP32 core updated with the master”. 
I have the latest JSON file for the ESP32. And the latest H libraries for Espressif. All of my downloads and updates have occurred in the last 2 weeks.

**Comment by pedrominatel** [comment]
I mean thay you'll only get this new feature in the next release and after you updating to the latest version. We'll release the new version mid December.

If you want to get the latest code before release, you need to update yourself, not using the JSON install process.

**Pull request merged**